Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2483,10 +2483,22 @@ bool RenderElement::hasEligibleContainmentForSizeQuery() const
     return false;
 }
 
+static void forEachDescendantInPostOrder(void (*function)(RenderObject&), RenderElement& parent)
+{
+    for (CheckedRef child : childrenOfType<RenderObject>(parent)) {
+        if (is<RenderElement>(child))
+            forEachDescendantInPostOrder(function, downcast<RenderElement>(child));
+        else
+            function(child);
+    }
+    function(parent);
+};
```

Comment:
It’s OK to do it in this functional style, although ideally if this comes up a lot we will want to add `descentsOfTypePostOrder` to `RenderDescendantIterator.h`.

But we want a non-recursive algorithm. Here’s one cut at that:
```
static RenderObject& firstInPostOrder(RenderObject& object)
{
    if (auto* firstLeafChild = next->firstLeafChild())
        return *firstLeafChild;
    return object;
}

static RenderObject* nextInPostOrder(RenderObject& object)
{
    if (auto* next = object->nextSibling())
        return &firstInPostOrder(*next);
    return object->parent();
}

static void forEachDescendantInPostOrder(void (*function)(RenderObject&), RenderElement& parent)
{
    for (auto* next = &firstInPostOrder(parent); next != &parent; next = nextInPostOrder(*next)) {
        function(*next);
    }
}
```

Reply:
@darinadler  I've implemented `descentsOfTypePostOrder` using your code snippet as a base. Thank you for the guidance!